engine/src/main/java/org/hibernate/search/engine/search/aggregation/dsl/AvgAggregationFieldStep.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.engine.search.aggregation.dsl;
+
+import java.util.function.Function;
+
+import org.hibernate.search.engine.search.common.ValueModel;
+import org.hibernate.search.engine.search.predicate.dsl.SearchPredicateFactory;
+
+/**
+ * The initial step in an "avg" aggregation definition, where the target field can be set.
+ *
+ * @param <PDF> The type of factory used to create predicates in {@link AggregationFilterStep#filter(Function)}.
+ */
+public interface AvgAggregationFieldStep<PDF extends SearchPredicateFactory> {
```

Comment:
could you please also go through these new DSL interfaces and mark them as incubating? You've already did it for the new methods in the factories, and we probably won't change these interfaces, but just to be on the safe side ... 🙈 😃 thanks!

Reply:
Sure!

Reply:
thanks!